packages/stark-ui/src/modules/table/components/table.component.ts

```diff
@@ -0,0 +1,595 @@
+
+	/**
+	 * Reference to the MatPaginator embedded in this component
+	 */
+	@ViewChild(MatPaginator) public paginator: MatPaginator;
+
+	/**
+	 * Columns added automatically by this component according to the columnProperties input
+	 */
+	@ViewChildren(StarkTableColumnComponent) public viewColumns: QueryList<StarkTableColumnComponent>;
+
+	/**
+	 * Columns added by the user via transclusion inside an <div> element with class "stark-table-columns"
+	 */
+	@ContentChildren(StarkTableColumnComponent) public contentColumns: QueryList<StarkTableColumnComponent>;
+
+	public columns: StarkTableColumnComponent[];
+	public selection: SelectionModel<any> = new SelectionModel<any>(true, []);
+
+	/**
+	 * MatTableDataSource associated to the MatTable embedded in this component
+	 */
+	public dataSource: MatTableDataSource<any>;
+
+	/**
+	 * Array of columns (column id's) to be displayed in the table.
+	 */
+	public displayedColumns: string[];
+
+	/**
+	 * Whether the sorting by multiple columns is enabled
+	 */
+	public isMultiSorting: boolean = false;
+
+	/**
+	 * Class constructor
+	 * @param dialogService - Angular Material service to open Material Design modal dialogs.
+	 * @param cdRef - Reference to the change detector attached to this component
+	 * @param logger - The logging service of the application
+	 */
+	public constructor(
+		public dialogService: MatDialog,
+		private cdRef: ChangeDetectorRef,
+		@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService
+	) {}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngOnInit(): void {
+		this.logger.debug(componentName + ": controller initialized");
+		this.displayedColumns = [];
+		if (this.multiselect) {
+			this.displayedColumns.unshift("select");
+		}
+	}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngAfterContentInit(): void {
+		this.logger.debug(componentName + ": ngAfterContentInit");
+	}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngAfterViewInit(): void {
+		this.logger.debug(componentName + ": ngAfterViewInit");
+		// add the columns the developer defined with the <stark-table-column>
+		this.columns = this.viewColumns.toArray().concat(this.contentColumns.toArray());
+
+		this.removeOldColumnsFromTable();
+		this.addColumnsToTable(this.columns);
+
+		if (this.orderProperties instanceof Array && this.orderProperties.length) {
+			// TODO: refactor -> sortData() internally calls initializeDataSource() and applyFilter()
+			this.sortData();
+		} else {
+			this.initializeDataSource();
+			this.applyFilter();
+		}
+
+		this.cdRef.detectChanges();
+	}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngOnChanges(changes: SimpleChanges): void {
+		if (changes["data"]) {
+			if (!changes["data"].isFirstChange()) {
+				if (this.resetFilterValueOnDataChange()) {
+					this.applyFilter();
+				}
+			}
+		}
+
+		if (changes["orderProperties"]) {
+			if (!changes["orderProperties"].isFirstChange()) {
+				this.sortData();
+			}
+		}
+
+		if (changes["filter"]) {
```

Comment:
Why? In this ngOnChanges we handle the updates of 3 inputs: data, orderProperties and filter. Whereas in the ColumnComponent we only handle 1 input: name. In any case, as I said in the other comment of the ColumnComponent, we might refactor the getter/setter in case we need it. But this ngOnChanges block should not change in my opinion